src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  // const group = {} as GroupsMap<T>;
+
+  const group = items.reduce((books: GroupsMap<T>, book) => {
+    const value = String(book[key]);
+
+    if (!books[value]) {
+      // eslint-disable-next-line no-param-reassign
```

Comment:
for avoid this elsint rule
you can assign books to some variable and work with this variable
or use ternay operator and destructuring
```
 return books[value] ? {
      ...books, [value]: [...books[value], book],
    } : {
      ...books, [value]: [book],
    };